### PR TITLE
rename hellworld-java to helloworld-java-spring

### PR DESCRIPTION
**Pull request by saturnism**
Fixes parts of #832
Making way for #1155

### CODE REVIEW

**Comment by saturnism** [comment]
@samodell the integration test is running against helloworld-java. Do you know which file I need to fix for the e2e test?

**Comment by samodell** [comment]
I do not, but @chaodaiG  should know.

**Comment by chaodaiG** [comment]
https://github.com/knative/docs/blob/master/test/sampleapp/config.yaml#L2

**Comment by saturnism** [comment]
@chaodaiG i thought that was it as well. would the language be "java-spring" then?

**Comment by chaodaiG** [comment]
@saturnism , `java-spring` should work

File: test/sampleapp/config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 languages:
-  - language: "java"
+  - language: "java-spring"
     expectedOutput: "Hello Spring Boot Sample v1!"
     preCommands:
       - exec: "curl"
```
Comment (chaodaiG) [inline test/sampleapp/config.yaml]: Sorry my bad, change 3 lines below is needed:
from: `args: "helloworld.zip -d helloworld-java_tmp"`
to:       `args: "helloworld.zip -d helloworld-java-spring_tmp"`

**Comment by chaodaiG** [comment]
@saturnism , there is another change missing, I added a comment in your PR

File: test/sampleapp/config.yaml (hunk 1 above)
Comment (saturnism) [reply test/sampleapp/config.yaml]: done.

**Comment by chaodaiG** [comment]
unfortunately you also need to replace `helloworld-java` with `helloworld-java-spring` in `serving.yaml`

**Comment by saturnism** [comment]
> unfortunately you also need to replace `helloworld-java` with `helloworld-java-spring` in `serving.yaml`

good catch! i've updated `service.yaml` and `README.md` to reflect this.

**Comment by saturnism** [comment]
quick ping.

**Pull request merged**